### ISSUE TITLE
Are there any requirements for porting Bao to the new platform？

### ISSUE THREAD

**Issue by Sunt-git** [bao-project/bao-hypervisor#18]
I want to port Bao to a new architecture platform, such as SPARC architecture, do I need any features of SPARC architecture, such as hardware virtualization extensions?

**Comment by josecm** [comment]
Porting Bao to a new platform on armv8-a or rv64h architectures is quite straightforward. However, porting to a completely new architecture is a different story. I am not familiar with SPARC, so it really depends the degree to which it resembles these architectures. I'd say if it implements traditional radix-tree page-tables, it is already half-way there. And yes, virtualization support (i.e. extra privilege mode + nested translation) is a requirement for porting bao. However, we have started working on [support for mpu-based architectures](https://github.com/josecm/bao-hypervisor/tree/wip/mpu-support) recently and we are decoupling the base virtualization functionalities from memory management and mapping. So, even if SPARC follows some other memory management architecture I'm not familiar with, it might be easier to incorporate it after this decoupling is finished.

Let me know if you are indeed planning on implementing a port for SPARC, I'd be glad to provide any assistance.

**Comment by Sunt-git** [comment]
Thanks for your reply, I have learned that SPARC architecture does not support the Hardware Virtualization Extensions, and Bao is implemented only in architectures which provide hardware-assisted virtualization,so porting Bao to SPARC may be a bit difficult.

I am learning Bao Hypervisor and hope to make some contributions to Bao in my postgraduate study.

**Comment by josecm** [comment]
> Thanks for your reply, I have learned that SPARC architecture does not support the Hardware Virtualization Extensions

That's unfortunate. Might I ask why you are targeting SPARC?

> I am learning Bao Hypervisor and hope to make some contributions to Bao in my postgraduate study.

Great! Hope to your from you soon!

**Comment by Sunt-git** [comment]
> That's unfortunate. Might I ask why you are targeting SPARC?
> Because I'm doing some work with SPARC to see if I can combine the two, and of course porting to other architectures is also possible, do you have any suggestions?

**Comment by Sunt-git** [comment]
> That's unfortunate. Might I ask why you are targeting SPARC?
Because I'm doing some work with SPARC to see if I can combine the two, and of course porting to other architectures is also possible, do you have any suggestions

**Comment by Sunt-git** [comment]
> > That's unfortunate. Might I ask why you are targeting SPARC?
> 
>Because I'm doing some SPARC-related work to see if the two works can be combined, of course, porting to other architectures is also possible, do you have any suggestions?

**Comment by josecm** [comment]
> Because I'm doing some SPARC-related work to see if the two works can be combined

Of course, that makes sense. Do you run other hypervisors without virtualization support on SPARC?

 > porting to other architectures is also possible, do you have any suggestions?

I'm not really that familiar with other architectures besides Arm and RISC-V. But x86 is always an obvious candidate. Also, I remember MIPS had a virtualization extension. However I'm not sure the extent to which MIPS is still used today, and which is the availability of the extensions on the available platforms.
